"flutter build ios-framework" fails if the attached iOS device is locked

## Steps to Reproduce

1. Attach a locked iPhone
2. Run `flutter build ios-framework` from a module

**Expected results:** <!-- what did you want to see? -->
Should succeed, build against the generic iOS destination and not a real device.

**Actual results:** <!-- what did you see? -->
```
 ├─Building plugins...                                              7.7s
Unable to build plugin frameworks: 2020-02-04 13:01:56.604 xcodebuild[9716:12763977] [MT] iPhoneConnect: 📱<DVTiOSDevice (0x7fe0e7e536d0), Jenn’s iPhone, iPhone, 13.3.1 (17D50), d83d5bc53967baa0ee18626ba87b6254b2ab5418> ==
Underlying device preparation errors ==
2020-02-04 13:01:56.604 xcodebuild[9716:12763977] [MT] iPhoneConnect: 📱<DVTiOSDevice (0x7fe0e7e536d0), Jenn’s iPhone, iPhone, 13.3.1 (17D50), d83d5bc53967baa0ee18626ba87b6254b2ab5418> -- Failed
_shouldMakeReadyForDevelopment check even though device is not locked by passcode.
Domain: com.apple.platform.iphoneos
Code: 5
Failure Reason: allowsSecureServices: 1. isConnected: 0. Platform:
<DVTPlatform:0x7fe127e0dab0:'com.apple.platform.iphoneos':<DVTFilePath:0x7fe127e0d990:'/Users/m/Applications/Xcode_11-3-1.app/Contents/Developer/Platforms/iPhoneOS.platform'>>. DTDKDeviceIdentifierIsIDID: 0
--
2020-02-04 13:01:56.604 xcodebuild[9716:12763977] [MT] iPhoneConnect: 📱<DVTiOSDevice (0x7fe0e7e536d0), Jenn’s iPhone, iPhone, 13.3.1 (17D50), d83d5bc53967baa0ee18626ba87b6254b2ab5418> == END: Underlying device preparation
errors ==
** BUILD FAILED **
```
```
$ flutter doctor -v

[✓] Flutter (Channel master, v1.14.7-pre.103, on Mac OS X 10.15.2 19C57, locale en-US)
    • Flutter version 1.14.7-pre.103 at /Users/m/Projects/flutter
    • Framework revision f14bc8fa5c (4 hours ago), 2020-02-04 11:36:41 -0500
    • Engine revision f44f50f3c5
    • Dart version 2.8.0 (build 2.8.0-dev.6.0 c8ed304e97)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/m/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Users/m/Applications/Xcode_11-3-1.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C505
    • CocoaPods version 1.8.4

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 42.1.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.41.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.7.1

[✓] Connected device (1 available)
    • Jenn’s iPhone • d83d5bc53967baa0ee18626ba87b6254b2ab5418 • ios • iOS 13.3.1

• No issues found!
```

--- a/packages/flutter_tools/lib/src/commands/build_ios_framework.dart
+++ b/packages/flutter_tools/lib/src/commands/build_ios_framework.dart
@@ -460,6 +460,7 @@ end
         'iphoneos',
         '-configuration',
         xcodeBuildConfiguration,
+        '-destination generic/platform=iOS',
         'SYMROOT=${iPhoneBuildOutput.path}',
         'BITCODE_GENERATION_MODE=$bitcodeGenerationMode',
         'ONLY_ACTIVE_ARCH=NO' // No device targeted, so build all valid architectures.
@@ -484,6 +485,7 @@ end
           'iphonesimulator',
           '-configuration',
           xcodeBuildConfiguration,
+          '-destination generic/platform=iOS',
           'SYMROOT=${simulatorBuildOutput.path}',
           'ARCHS=x86_64',
           'ONLY_ACTIVE_ARCH=NO'

